wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -10,21 +14,60 @@
 
     <div class="help-block help-info">
         {% icon name="help" %}
-        This action will save a new page revision.
+        These actions will save a new page revision.
     </div>
 
     <p>
+        <input
+            type="hidden"
+            name="newsletter-test-email"
+            data-wn-panel-target="testAddress"
+        >
+
         <button
             type="submit"
             class="button button-secondary button-longrunning"
-            name="newsletter_action"
+            name="newsletter-action"
             value="save_campaign"
             data-controller="w-progress"
             data-action="w-progress#activate"
         >
             {% icon name="spinner" %}
             Save campaign to {{ backend_name }}
         </button>
+
+        <button
+            type="button"
+            class="button button-secondary button-longrunning"
+            data-a11y-dialog-show="wn-test-dialog"
+            data-controller="w-progress"
+            data-wn-panel-target="testButton"
+        >
+            {% icon name="spinner" %}
+            Send test email
+        </button>
+
+        {% dialog icon_name="mail" id="wn-test-dialog" title="Test email" %}
+            <form
+                data-controller="wn-test"
+                data-action="wn-test#send:prevent"
+            >
+                {% include "wagtailadmin/shared/field.html" with field=test_form.email %}
+
+                <button type="submit" class="button button-primary">Send</button>
+                <button type="button" class="button button-secondary" data-action="w-dialog#hide">Cancel</button>
+            </form>
+        {% enddialog %}
+
+        <button
+            type="submit"
+            class="w-hidden"
+            name="newsletter-action"
+            value="send_test_email"
+            data-wn-panel-target="testSubmit"
+        >
+            Send test email
+        </button>
```

Comment:
This is a hidden submit button, separate from the visible button (that only opens the dialog). Its purpose is to submit the form while including an extra field, newsletter_action=send_test_email. We could have used a hidden form input, but if for some reason the form submission got cancelled (e.g. by some other custom logic), that input would have stayed in the form, leading to unexpected results if the user saved the form normally.